### PR TITLE
table.exists() implemented. Unit test added for exists()

### PR DESCRIPTION
**Pull request by vikas-jamdar**
table.exist() implemented. Unit test added for exist()

### CODE REVIEW

File: bigtable/google/cloud/bigtable/table.py [hunk 1]
```diff
@@ -184,6 +184,16 @@ def create(self):
                                                 table_id=self.table_id,
                                                 table={})
 
+    def exist(self):
```
Comment (tseaver) [inline bigtable/google/cloud/bigtable/table.py]: Our convention would be to spell this method name `exists`. The docstring should be in the form:

```python
    def exists(self):
        """Check whether the table exists.

        :rtype: bool
        :returns: True if the table exists, else False.
        """"
```

File: bigtable/google/cloud/bigtable/table.py [hunk 2]
```diff
@@ -184,6 +184,16 @@ def create(self):
...
+        """Checks whether a table with given table ID exist
+        or not for given Bigtable instance.
+        """
```
Comment (tseaver) [inline bigtable/google/cloud/bigtable/table.py]: A cleaner / faster implementation is to use the table admin API `get_table` method, e.g.:

```python
        table_client = self._instance._client.table_admin_client
        instance_name = self._instance.name
        try:
            table_client.get_table(name=self.name)
        except exceptions.NotFound:
            return False
        else:
            return True
```

File: bigtable/google/cloud/bigtable/table.py [hunk 3]
```diff
@@ -184,15 +185,21 @@ def create(self):
...
-            if self.table_id == table.table_id:
-                return True
-        return False
+        table_client = self._instance._client.table_admin_client
+        try:
+            table_client.get_table(name=self.name)
+        except NotFound:
+            return False
+        except StopIteration:
+            return False
```
Comment (tseaver) [inline bigtable/google/cloud/bigtable/table.py]: I don't see how `google.cloud.bigtable_admin_v2.gapic.bigtable_table_admin_client.BigtableTableAdminClient.get_table` would ever raise a `StopIteration`:  it is defined to return a single `google.cloud.bigtable_admin_v2.types.Table` message.

File: bigtable/tests/unit/test_table.py [hunk 4]
```diff
@@ -283,6 +283,47 @@ def _create_test_helper(self):
     def test_create(self):
         self._create_test_helper()
 
+    def test_exist(self):
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Please match the method name change above, i.e., `test_exists`.

File: bigtable/tests/unit/test_table.py [hunk 5]
```diff
@@ -283,6 +283,47 @@ def _create_test_helper(self):
...
+                table_data_v2_pb2.Table(name=self.TABLE_NAME),
+            ],
+        )
+
+        # Patch API calls
+        client._table_admin_client = table_api
+        client._instance_admin_client = instance_api
+        bigtable_table_stub = (
+            client._table_admin_client.bigtable_table_admin_stub)
+        bigtable_table_stub.ListTables.side_effect = [response_pb]
```
Comment (tseaver) [inline bigtable/tests/unit/test_table.py]: Use a `NotFound` exception instance for the second call, e.g.:

```python
        from google.cloud.api_core import exceptions
        ...
        bigtable_table_stub.ListTables.side_effect = [
            response_pb,
            exceptions.NotFound('testing'),
        ]
```

File: bigtable/google/cloud/bigtable/table.py (hunk 3 above)
Comment (vikas-jamdar) [reply bigtable/google/cloud/bigtable/table.py]: Agree with you. I also worried about that. But get_table gives StopIteration. Can you check my unittest test_exists() populating mock environment correctly.

**Pull request merged**